gnosis/gp-swap-ui
# Release/0.10

## Do not Squash 🥒, just normal merge

```
# To get commits used in the description (useful for next releases) 
git log master..HEAD --oneline --decorate=0

git log v0.8..HEAD --oneline --decorate=0
```

# 💪Work In Progress
🚨This version is not ready to be merged!!🚨 
Pending to incorporate some open PRs:
* ~~The ETH/WETH workflow~~
* ~~FAQ~~
* ~~About page~~
* We need to remove the policies, and setup ENS deployment
* Final touches 🪄✨



# Release notes (QA instructions)
Since we didn't test version 0.9, I will include the notes of both v0.9 and v0.10

## v0.10
###  Faq (#470)
There's a new page with the FAQ accessible from the menu in the header:
![image](https://user-images.githubusercontent.com/2352112/115578513-a7a81680-a2c5-11eb-9451-66ea49842a3b.png)

Instructions to test:
1. Navigate to the page
2. Review the content

### About page #459
There's a new about page.

![image](https://user-images.githubusercontent.com/2352112/115577857-0a4ce280-a2c5-11eb-9d8f-0b967d8c430d.png)

Instructions to test:
1. Navigate to the page
2. Review the content

### Click wrap (#474)
On new component that would allow the user to accept the cookie policy.

This component is not fully implemented, in this version only the styles and the component were implemented. 
It is hidden (we don't have clickwrap in this version).

Instructions to test:
* No instruction :), nothing to test for now. 


### ETH/WETH workflow #462
Improve UX for selling ETH.

Cow Swap doesn't allow you to sell ETH. This is why it needs to be wrapped first. 
This version includes a little help to make it easier to wrap you ETH.


Instructions to test:
1. Select ETH as the sell token
2. Select any other token as buy token
3. Input amount
4. Notice the user can now only do two things: 1. Select WETH (main button), 2. Wrap ETH into WETH
5. Test that both buttons work as expected, and that the UX is indeed better (more explicit and guided)

### Fee re-fetch improvements (#475)
Closes a couple of issues (Closes #456, Closes #472) and improves the fee handling. 

With this change, we makes sure we don't re-fetch the fee if the quote parameters don't change
* ...unless, the quote is about to expire (<30 seconds)
* It also adds a periodic check (every 15seconds)

Instructions to test:
1. Open developer tools, Nework tab https://developer.chrome.com/docs/devtools/network/
2. Select a token pair
3. Input amount
4. Observe there's a request to the fee endpoint with the right parameters
5. Continue changing the amount, and tokens and see that there's new request with the new parameters
6. Don't make any changes in the form, wait until fee is expiring in 30s (it takes around 30s to happen)

![image](https://user-images.githubusercontent.com/2352112/115526180-10749c00-a290-11eb-9508-ac65692ee2f5.png)


## Use production Explorer link for explorer
#441

> Unfortunately this change can only be tested in `staging` branch, and production. @MareenG when we are closer to the launch, please ask us to deploy this to staging/production to verify

The app will use different links for the explorer depending if we are in:
* DEVELOPMENT: https://protocol-explorer.dev.gnosisdev.com
* STAGING: https://protocol-explorer.staging.gnosisdev.com
* PRODUCTION: https://gnosis-protocol.io

Instructions to test:
1. Create a new trade
2. Follow the link to the explorer
3. Verify is the right link

### Buy ETH #451
> @MareenG for testing this one, you still need to wait a little bit. The Backend is working on finishing this functionality. @fleupold can update you when this is testable. 

Instructions in the PR

### Fix price impact display
Instructions in the PR #457

### Add kimono cow in advance settings
Instructions to test:
1. Enable advanced settings
2. Observe kimono-cow! 🐮🥋

###  Added an alpha indicator to the logo #385
![image](https://user-images.githubusercontent.com/2352112/115528529-68ac9d80-a292-11eb-9501-17ab4dcc6fa0.png)

### Diable all wallets but Metamask #430
All wallets but metamask should be disabled

Test instructions:
1. Click on connect
2. Metamask is the only option

### Wallet Connect connector #436 and #435

> No need to test this one, since Wallet connect has been disabled

Fixes an issue  with reading balances with WalletConnect

### Adds default App Id #447
Now the app id is 1. This is App id is just some information we send when we sign an order so we give some additional information with it.

Test instructions:
1. Trade (get to the signing modal that shows the data of the order to sign)
2. Observe that the app id is 0x1 in HEX

### Update to uniswap v3.2.14 #423
Small fixes. Details in #423

### Replace message: Operator was renamed to Network for some error messages #423
See changes https://github.com/gnosis/gp-swap-ui/pull/410/files


### Send tokens to a different address #412
Allows you to send the received tokens to a different address that the one used for trading

Test instructions:
1. Enable "Advanced" in the settings
2. Notice that you now can set a recipient address
3. Input a different address than the one you are trading with
4. Sign the order. Observe that the specified address is now part of the signed data
5. Sign and wait for the trade to execute. Make sure the specified account received the tokens

## v0.9
Details in #408

# Changelog

> 🚨Pending to add the latest changes
>  `git log 86cea66e..HEAD --oneline --decorate=0`

## v0.10
e6a19502 About page content (#459)
33670804 Click wrap Component + styles. (#474)
b2ca009c [443] - ETH/WETH Flow - Base PR (#462)
86cea66e Fee re-fetch improvements (#475)
b0d8c9ea Fix price impact display (#457)
37112496 Adding Settings label + expert 🐮 (#440)
f8f7af7d Add new official url (#453)
0e77e9b9 [421] Environment-aware Explorer URLs (#441)
40124702 ETH can be used as BUY token (#451)
ef303cfe Wallet connect connector (#436)
ed66ad90 Improve logo app status/header component. (#385)
8d441c06 Disable not-fully working wallets (#430)
46613892 Add multi-network support (#435)
4014df2c Add default app-id (#447)
8478ee5b Merge remote-tracking branch 'origin/master' into develop
4a62b4f4 Update uni v3.2.14 (#423)
5ae56bbb Merge remote-tracking branch 'origin/release/v0.9.0' into develop
dc164dbf Merge pull request #414 from gnosis/fleupold-patch-1
b9bc216a Merge remote-tracking branch 'origin/release/v0.9.0' into develop
8336e3d0 Update README.md
38a4c22b Replace `operator` by `network` on unknown error message (#410)
9da6903c Re-enable adding/removing send address (#412)

## v0.9
Details in #408
